<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
  <class>MainWindow</class>
  <widget class="QMainWindow" name="MainWindow">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>800</width>
        <height>600</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>Webcam Viewer</string>
    </property>
    <property name="styleSheet">
      <string notr="true">background-color: #353535;</string>
    </property>
    <widget class="QWidget" name="centralwidget">
      <layout class="QVBoxLayout" name="mainLayout">
        <!-- Верхняя панель с кнопками -->
        <item>
          <widget class="QWidget" name="topPanel">
            <property name="styleSheet">
              <string notr="true">background-color: transparent;</string>
            </property>
            <layout class="QHBoxLayout" name="topPanelLayout">
              <item>
                <spacer name="horizontalSpacer">
                  <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                    <size>
                      <width>0</width>
                      <height>20</height>
                    </size>
                  </property>
                </spacer>
              </item>
              <item>
                <widget class="QPushButton" name="closeButton">
                  <property name="minimumSize">
                    <size>
                      <width>30</width>
                      <height>30</height>
                    </size>
                  </property>
                  <property name="maximumSize">
                    <size>
                      <width>30</width>
                      <height>30</height>
                    </size>
                  </property>
                  <property name="text">
                    <string>×</string>
                  </property>
                  <property name="styleSheet">
                    <string notr="true">QPushButton {
                      background-color: #ff5c5c;
                      color: white;
                      border-radius: 15px;
                      border: none;
                      font-size: 18px;
                      font-weight: bold;
                    }
                    QPushButton:hover {
                      background-color: #ff3b3b;
                    }
                    QPushButton:pressed {
                      background-color: #ff1a1a;
                    }</string>
                  </property>
                </widget>
              </item>
            </layout>
          </widget>
        </item>

        <!-- Основной контент -->
        <item>
          <widget class="QLabel" name="cameraLabel">
            <property name="minimumSize">
              <size>
                <width>640</width>
                <height>480</height>
              </size>
            </property>
            <property name="styleSheet">
              <string notr="true">background-color: black;</string>
            </property>
            <property name="alignment">
              <set>Qt::AlignCenter</set>
            </property>
          </widget>
        </item>

        <!-- Панель управления -->
        <item>
          <layout class="QVBoxLayout" name="controlLayout">
            <item>
              <widget class="QLabel" name="sensitivityLabel">
                <property name="styleSheet">
                  <string notr="true">color: white; font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>Sensitivity: Medium</string>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QSlider" name="sensitivitySlider">
                <property name="minimum">
                  <number>0</number>
                </property>
                <property name="maximum">
                  <number>5</number>
                </property>
                <property name="value">
                  <number>2</number>
                </property>
                <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                </property>
                <property name="styleSheet">
                  <string notr="true">QSlider::groove:horizontal {
                    height: 8px;
                    background: #555;
                    border-radius: 4px;
                  }
                  QSlider::handle:horizontal {
                    width: 18px;
                    margin: -5px 0;
                    background: #4CAF50;
                    border-radius: 9px;
                  }</string>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QPushButton" name="startButton">
                <property name="minimumSize">
                  <size>
                    <width>120</width>
                    <height>30</height>
                  </size>
                </property>
                <property name="text">
                  <string>Start</string>
                </property>
                <property name="styleSheet">
                  <string notr="true">QPushButton {
                    background-color: #4CAF50;
                    color: white;
                    border: none;
                    padding: 5px 10px;
                    border-radius: 4px;
                    font-weight: bold;
                  }
                  QPushButton:hover {
                    background-color: #45a049;
                  }
                  QPushButton:pressed {
                    background-color: #367d39;
                  }</string>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QLabel" name="fpsLabel">
                <property name="styleSheet">
                  <string notr="true">background-color: rgba(0,0,0,50%);
                  color: white; padding: 5px;</string>
                </property>
                <property name="text">
                  <string>FPS: 0.0</string>
                </property>
              </widget>
            </item>
            <item>
              <widget class="QLabel" name="motionLabel">
                <property name="styleSheet">
                  <string notr="true">color: white; font-weight: bold;</string>
                </property>
                <property name="text">
                  <string>Motion Detected: No</string>
                </property>
              </widget>
            </item>
          </layout>
        </item>
      </layout>
    </widget>

    <!-- Меню -->
    <widget class="QMenuBar" name="menubar">
      <property name="geometry">
        <rect>
          <x>0</x>
          <y>0</y>
          <width>800</width>
          <height>22</height>
        </rect>
      </property>
      <widget class="QMenu" name="menuFile">
        <property name="title">
          <string>File</string>
        </property>
        <addaction name="actionExit"/>
      </widget>
      <widget class="QMenu" name="menuSettings">
        <property name="title">
          <string>Settings</string>
        </property>
        <addaction name="actionPreferences"/>
      </widget>
      <widget class="QMenu" name="menuHelp">
        <property name="title">
          <string>Help</string>
        </property>
        <addaction name="actionAbout"/>
      </widget>
      <addaction name="menuFile"/>
      <addaction name="menuSettings"/>
      <addaction name="menuHelp"/>
    </widget>

    <!-- Действия -->
    <action name="actionExit">
      <property name="text">
        <string>Exit</string>
      </property>
    </action>
    <action name="actionAbout">
      <property name="text">
        <string>About</string>
      </property>
    </action>
    <action name="actionPreferences">
      <property name="text">
        <string>Preferences</string>
      </property>
    </action>
  </widget>
  <resources/>
  <connections>
    <connection>
      <sender>actionExit</sender>
      <signal>triggered()</signal>
      <receiver>MainWindow</receiver>
      <slot>close()</slot>
    </connection>
  </connections>
</ui>

